ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -17,15 +17,39 @@ open Datamodel_types
 open CommonFunctions
 module Types = Datamodel_utils.Types
 
+type fields = string * field list
+
 let templates_dir = "templates"
 
 let ( // ) = Filename.concat
 
+let special_words =
+  [
+    "id"
+  ; "ip"
+  ; "vm"
+  ; "api"
+  ; "uuid"
+  ; "cpu"
+  ; "tls"
+  ; "https"
+  ; "url"
+  ; "db"
+  ; "xml"
+  ; "eof"
+  ]
+
 let snake_to_camel (s : string) : string =
-  Astring.String.cuts ~sep:"_" s
-  |> List.map (fun s -> Astring.String.cuts ~sep:"-" s)
+  String.split_on_char '_' s
+  |> List.map (fun s -> String.split_on_char '-' s)
   |> List.concat
```

Comment:
You can simplify this to use `List.concat_map` (which I believe to be tail recursive).

The function itself can also be η-reduced, but I'm unsure if that style is common in this codebase, e.g. `List.concat_map (String.split_on_char '-')`. It's [point-free style] nice in small doses but can have problems in other contexts; such as`let`-generalisation (w.r.t relaxed value restriction and weak type variables).

Similarly, below (lines `46`-`47`) there is the pattern of `fun s -> match s with ..` which is captured by the `function ...` form. Again, I am unsure how the formatter will look with this (it may indent it an extra level).

Please let me know if this it too pedantic or opinionated.

Reply:
Thanks. These are good suggestions.